### PR TITLE
Fix selector specificity for nested components

### PR DESCRIPTION
**Pull request by vvalentin-lucca**
## Description

SCSS components can be loaded via an explicit CSS call, but also via NG components.

As the order in which NG components were called could not be controlled, we ended up with style overloads when components were nested and overloaded within other components.

In this PR, I increase the specificity of selectors when this happens, to ensure that the order in which CSS components are called up no longer interferes. 

-----

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-3064/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/22500/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-3064/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/22501/ui-diff)

**Review by nsarradin — DISMISSED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-3064/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/22511/ui-diff)

**Review by Supamiu — APPROVED** [review]

File: packages/scss/src/components/dialog/component.scss [hunk 1]
```diff
@@ -27,18 +27,28 @@
 	flex-direction: column;
 
 	.dialog-inside-header-button {
-		@include button.text;
-		@include button.S;
-		@include button.onlyIconS;
+		// :not(.class) is only there to increase specificity when the class isn’t present
+		// but the class should be present, and this code is temporary
```
Comment (fbasmaison-lucca) [inline packages/scss/src/components/dialog/component.scss:31]: If the code is temporary, maybe add a `TODO` or `FIXME` prefix to the comment?

**Review by fbasmaison-lucca — APPROVED** [review]
Just a comment about comments, but thanks anyhow. :)

**Pull request merged**